pkg/proxy/iptables/proxier.go

```diff
@@ -1281,12 +1280,14 @@ func (proxier *Proxier) syncProxyRules() {
 					klog.V(4).InfoS("Port was open before and is still needed", "port", lp.String())
 					replacementPortsMap[lp] = proxier.portsMap[lp]
 				} else if svcInfo.Protocol() != v1.ProtocolSCTP {
-					socket, err := proxier.portMapper.OpenLocalPort(&lp, isIPv6)
+					socket, err := proxier.portMapper.OpenLocalPort(&lp)
 					if err != nil {
 						klog.ErrorS(err, "can't open port, skipping this nodePort", "port", lp.String())
 						continue
 					}
-					if lp.Protocol == "udp" {
+					klog.V(2).InfoS("Opened local port", "port", lp.String())
+
+					if lp.Protocol == utilnet.UDP {
```

Comment:
so, this was the problem, right?
we weren´t cleaning the conntrack entries and the test failed :thinking:

Reply:
I think it cause that. The difference of them is here. The code before, this if judge dont take effect.